javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -0,0 +1,103 @@
+/**
+ * @name Use of returnless function.
+ * @description Using the return value of a function that does not explicitly return is indicative of a mistake.
```

Comment:
I don't understand what a `function that doesn't explicitly return` is. What do we expect it to return? I am not a developer, and I'm just wondering whether readers will get this (if the answer is yes, feel free to ignore this comment).

Reply:
I think readers familiar with JavaScript development will get it. It could perhaps be changed to `explicitly return a value`? But that change would have in multiple locations.
@esben-semmle ?

Reply:
How about:

> Using the return value of a function that does not return an expression is indicative of a mistake.

Then there is no confusing implicit/explicit concept, and it is made clear that the `return` is a syntactic thing as it mentions an expression rather than a value.

Reply:
I'm using `Using the return value of a function that does not return an expression is indicative of a mistake.` for now.  
And then the users of the query can be happy that we remove the FP's from generators and async functions.